extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeCommentsExtractor.java

```diff
@@ -0,0 +1,263 @@
+					} catch (Exception e) {
+						throw new ParsingException("Could not get thumbnail url", e);
+					}
+				}
+
+				@Override
+				public Integer getLikeCount() throws ParsingException {
+					try {
+						return c.get("likeCount").intValue();
+					} catch (Exception e) {
+						throw new ParsingException("Could not get thumbnail url", e);
+					}
+				}
+
+				@Override
+				public String getCommentText() throws ParsingException {
+					try {
+						if (null != c.get("contentText").get("simpleText")) {
+							return c.get("contentText").get("simpleText").asText();
+						} else {
+							return c.get("contentText").get("runs").get(0).get("text").asText();
+						}
+					} catch (Exception e) {
+						throw new ParsingException("Could not get thumbnail url", e);
+					}
+				}
+
+				@Override
+				public String getCommentId() throws ParsingException {
+					try {
+						return c.get("commentId").asText();
+					} catch (Exception e) {
+						throw new ParsingException("Could not get thumbnail url", e);
+					}
+				}
+
+				@Override
+				public String getAuthorThumbnail() throws ParsingException {
+					try {
+						return c.get("authorThumbnail").get("thumbnails").get(0).get("url").asText();
+					} catch (Exception e) {
+						throw new ParsingException("Could not get thumbnail url", e);
+					}
+				}
+
+				@Override
+				public String getAuthorName() throws ParsingException {
+					try {
+						return c.get("authorText").get("simpleText").asText();
+					} catch (Exception e) {
+						throw new ParsingException("Could not get thumbnail url", e);
+					}
+				}
+
+				@Override
+				public String getAuthorEndpoint() throws ParsingException {
+					try {
+						return "https://youtube.com"
+								+ c.get("authorEndpoint").get("browseEndpoint").get("canonicalBaseUrl").asText();
+					} catch (Exception e) {
+						throw new ParsingException("Could not get thumbnail url", e);
+					}
+				}
+			};
+
+			collector.commit(extractor);
+		});
+
+	}
+
+	@Override
+	public void onFetchPage(Downloader downloader) throws IOException, ExtractionException {
+		DownloadResponse response = downloader.downloadWithHeaders(getUrl());
+		String responseBody = response.getResponseBody();
+		cookies = response.getResponseHeaders().get("Set-Cookie");
+		sessionToken = findValue(responseBody, "XSRF_TOKEN");
+		commentsToken = findValue(responseBody, "COMMENTS_TOKEN");
+	}
+
+	@Override
+	public String getName() throws ParsingException {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	private String makeAjaxRequest(String siteUrl) throws IOException {
+
+		StringBuilder postData = new StringBuilder();
+		postData.append(URLEncoder.encode("session_token", "UTF-8"));
+		postData.append('=');
+		postData.append(URLEncoder.encode(sessionToken, "UTF-8"));
+		byte[] postDataBytes = postData.toString().getBytes("UTF-8");
+
+		URL url = new URL(siteUrl);
+		HttpsURLConnection con = (HttpsURLConnection) url.openConnection();
```

Comment:
No HttpsURLConnection. We use `Downloader` for this purpouse, in order to wrap the Network library.

Reply:
Since the `Downloader` does not yet support post requests should I do a PR for just that first as suggested in this comment https://github.com/TeamNewPipe/NewPipeExtractor/pull/75#issuecomment-371985858 by @karyogamy . 
Quoting from the same comment
> This may impact how list extraction passes the next url info since we now need to take form-data into account. In particular. With this in mind, maybe we should consider passing around a data class rather than a String on getNextPageUrl()?